src/util/format_expr.cpp

```diff
@@ -128,10 +129,18 @@ static std::ostream &format_rec(
     else
       return os << src.pretty();
   }
-  else if(type == ID_unsignedbv || type == ID_signedbv)
+  else if(type == ID_unsignedbv ||
+          type == ID_signedbv ||
+          type == ID_c_bool)
     return os << *numeric_cast<mp_integer>(src);
   else if(type == ID_integer)
     return os << src.get_value();
+  else if(type == ID_string)
+  {
+    os << '"';
+    jsont::escape_string(id2string(src.get_value()), os);
+    return os << '"';
+  }
   else if(type == ID_floatbv)
```

Comment:
Because I've just run into it: could
````c++
+  else if(type == ID_pointer && (src.is_zero() || src.get_value() == ID_NULL))
+    return os << src.get_value();
````
please be added as well?

Reply:
Done (src.is_zero() should suffice)